HW0/P4/P4.py

```diff
@@ -30,47 +30,92 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    s_true=np.loadtxt("P4_trajectory.txt", delimiter=",")
+    x_coords=s_true[:,0]
+    y_coords=s_true[:,1]
+    z_coords=s_true[:,2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+            '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    m = np.loadtxt("P4_measurements.txt", delimiter=",")
+    C = np.array([[1./rx, 0.0, 0.0],
+                  [0.0, 1./ry, 0.0],
+                  [0.0, 0.0, 1./rz]])
+    s_obs=np.dot(C, m.T)
+    
+    x_coords=s_obs[0]
+    y_coords=s_obs[1]
+    z_coords=s_obs[2]
+
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.array([[1.0, 0.0, 0.0, dt, 0.0, 0.0],
```

Comment:
np.diag() (and other numpy functions) can make making these matricies a little more clean.

Of course, it's very clear what this matrix is in this form, so just giving a heads up. :)